DataBiosphere/leonardo
# [IA-4363] Adding Project Permission Check to V1 endpoints

__Jira ticket__: https://broadworkbench.atlassian.net/browse/[ticket_number]




## Summary of changes



### What

- Adding project-level Sam checks to V1 endpoints:
- Disks (get, list, delete, update)
- Apps (get, list, delete, start, stop)
- Runtimes (get, list, delete, start, stop, update)

### Why

- Users who are removed from projects shouldn't be able to see or interact with those resources

## Testing these changes

### What to test

- Add a user to a project
- Have them create an app and a runtime
- Have them get, list, start, stop, update successfully
- Remove the user from that workspace
- Attempt to get, list (both by project and listAll), start, stop, update the disk/runtime/app
- Should return a 403 or an empty list

### Who tested and where

  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
